actors/builtin/miner/miner_state_test.go

```diff
@@ -440,6 +442,252 @@ func TestVestingFundsStore(t *testing.T) {
+		assert.True(t, harness.vestingFundsStoreEmpty())
+	})
+
+	t.Run("Vests when quantize, step duration, and vesting period are coprime", func(t *testing.T) {
+		harness := constructStateHarness(t, abi.ChainEpoch(0))
+		vspec := &miner.VestSpec{
+			InitialDelay: 0,
+			VestPeriod:   27,
+			StepDuration: 5,
+			Quantization: 7,
+		}
+		vestStart := abi.ChainEpoch(10)
+		vestSum := abi.NewTokenAmount(100)
+		harness.addLockedFunds(vestStart, vestSum, vspec)
+		// Entry: 21 Funds: 40
+		// Entry: 28 Funds: 26
+		// Entry: 35 Funds: 26
+		// Entry: 42 Funds: 8
+		totalVested := abi.NewTokenAmount(0)
+		for e := vestStart; e <= 43; e++ {
+			amountVested := harness.unlockVestedFunds(e)
+			switch e {
+			case 22:
+				assert.Equal(t, abi.NewTokenAmount(40), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			case 29:
+				assert.Equal(t, abi.NewTokenAmount(26), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			case 36:
+				assert.Equal(t, abi.NewTokenAmount(26), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			case 43:
+				assert.Equal(t, abi.NewTokenAmount(8), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			default:
+				assert.Equal(t, abi.NewTokenAmount(0), amountVested)
+			}
+		}
+		assert.Equal(t, vestSum, totalVested)
+		assert.Zero(t, harness.s.LockedFunds.Int64())
+		assert.True(t, harness.vestingFundsStoreEmpty())
+	})
+
+	t.Run("Vests alternating amounts", func(t *testing.T) {
+		harness := constructStateHarness(t, abi.ChainEpoch(0))
+		vspec := &miner.VestSpec{
+			InitialDelay: 0,
+			VestPeriod:   100,
+			StepDuration: 9,
+			Quantization: 2,
+		}
+		vestStart := abi.ChainEpoch(100)
+		vestSum := abi.NewTokenAmount(1000)
+		harness.addLockedFunds(vestStart, vestSum, vspec)
+		/*
+			Entry: 110 Funds: 100
+			Entry: 118 Funds: 80
+			Entry: 128 Funds: 100
+			Entry: 136 Funds: 80
+			Entry: 146 Funds: 100
+			Entry: 154 Funds: 80
+			Entry: 164 Funds: 100
+			Entry: 172 Funds: 80
+			Entry: 182 Funds: 100
+			Entry: 190 Funds: 80
+			Entry: 200 Funds: 100
+		*/
+		// mapping of epochs to value vested at it, epochs are all incremented by 1 since vesting occurs after them
+		vestEpochs := map[abi.ChainEpoch]int64{
+			111: 100, 119: 80, 129: 100, 137: 80, 147: 100,
+			155: 80, 165: 100, 173: 80, 183: 100, 191: 80, 201: 100,
+		}
+		totalVested := abi.NewTokenAmount(0)
+		for e := vestStart; e < vestStart+vspec.VestPeriod+2; e++ {
+			amountVested := harness.unlockVestedFunds(e)
+			if a, vests := vestEpochs[e]; vests {
+				assert.Equal(t, abi.NewTokenAmount(a), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			} else {
+				assert.Equal(t, abi.NewTokenAmount(0), amountVested)
+			}
+		}
+		assert.Equal(t, vestSum, totalVested)
+		assert.Zero(t, harness.s.LockedFunds.Int64())
+		assert.True(t, harness.vestingFundsStoreEmpty())
+	})
+
+	t.Run("Unlock unvested funds leaving bucket with modified amount", func(t *testing.T) {
```

Comment:
Move tests primarily about _un_vested funds to a different method. All the complex specs are pretty much irrelevant to them.